compiler/rustc_passes/src/check_attr.rs

```diff
@@ -2368,18 +2391,33 @@ impl<'tcx> CheckAttrVisitor<'tcx> {
                 }
             }
 
-            infcx.err_ctxt().note_type_err(
-                &mut diag,
-                &cause,
-                None,
-                Some(ValuePairs::PolySigs(ExpectedFound {
-                    expected: ty::Binder::dummy(expected_sig),
-                    found: ty::Binder::dummy(sig),
-                })),
-                terr,
-                false,
-                false,
-            );
+            let mut note_expected_found = |expected_sig| {
+                infcx.err_ctxt().note_type_err(
+                    &mut diag,
+                    &cause,
+                    None,
+                    Some(ValuePairs::PolySigs(ExpectedFound {
+                        expected: ty::Binder::dummy(expected_sig),
+                        found: ty::Binder::dummy(sig),
+                    })),
+                    terr,
+                    false,
+                    false,
+                )
+            };
+
+            note_expected_found(expected_sig);
+
+            if let ProcMacroKind::Derive = kind
+                && tcx
+                    .features()
+                    .declared_lib_features
+                    .iter()
+                    .any(|&(feature, _)| feature == sym::derive_const)
```

Comment:
I think I should actually be able to do:

```suggestion
                && tcx.features().declared(sym::derive_const)
```

or is it

```suggestion
                && tcx.features().active(sym::derive_const)
```